spec/registrant_spec.rb

```diff
@@ -0,0 +1,40 @@
+require './lib/registrant'
+
+RSpec.describe Registrant do 
+  describe "#initialize" do 
+    it "exists" do
+      registrant_1 = Registrant.new('Bruce', 18, true )
+      registrant_2 = Registrant.new('Penny', 15 )
+      
+      expect(registrant_1).to be_a Registrant
+      expect(registrant_2).to be_a Registrant
+    end
+    
+    it "has an age, name and permit" do 
+      registrant_1 = Registrant.new('Bruce', 18, true )
+      registrant_2 = Registrant.new('Penny', 15 )
+      
+      
+      expect(registrant_1.name).to eq("Bruce")
+      expect(registrant_1.age).to eq(18)
+      expect(registrant_1.permit?).to eq(true)
+      expect(registrant_2.name).to eq("Penny")
+      expect(registrant_2.age).to eq(15)
+      expect(registrant_2.permit?).to eq(false)
+    end
+    
+    it "has liscense data" do 
+      registrant_2 = Registrant.new('Penny', 15 )
+      
+      expect(registrant_2.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+    end 
+    
+    it "can earn a permit" do 
+      registrant_2 = Registrant.new('Penny', 15 )
+
+      registrant_2.earn_permit
+
+      expect(registrant_2.permit).to eq(true)
```

Comment:
The reason this passes the test despite missing the `?` is because you put `permit` in your attr_reader. It's often tempting at first to put every attribute in an attr_reader out of habit, but you generally don't want to put anything in an attr_reader unless you have to. In the case of this project, your method `permit?` already covers that attribute, so it doesn't need to be in the attr_reader.

You'll come across the concept of "YAGNI" or "You Ain't Gonna Need It." Basically means "only write the minimum amount of code you actually need."